chalicelib_smaht/checks/wfr_checks.py

```diff
@@ -143,22 +153,25 @@ def md5runsmaht_status(connection, file_type="", start_date=None, **kwargs):
         else:  # covers cases of SubmittedFile, ReferenceFile
             my_bucket = raw_bucket
         # check if file is in s3
-        file_id = a_file["accession"]
+        file_id = a_file["uuid"]
         head_info = my_s3_util.does_key_exist(a_file["upload_key"], my_bucket)
         if not head_info:
             no_s3_file.append(file_id)
             continue
-        md5_report = wfr_utils.get_wfr_out(a_file, "md5", key=my_auth, md_qc=True)
-        if md5_report["status"] == "running":
-            running.append(file_id)
-        elif md5_report["status"].startswith("no complete run, too many"):
-            problems.append(file_id)
-        # most probably the trigger did not work, and we run it manually
-        elif md5_report["status"] != "complete":
+        md5_mwfrs_for_file = get_md5_mwfrs_for_file(my_auth, a_file["uuid"], md5_mwf["uuid"])
+        if len(md5_mwfrs_for_file) == 0:
             missing_md5.append(file_id)
-        # there is a successful run, but status is not switched, happens when a file is reuploaded.
-        elif md5_report["status"] == "complete":
-            not_switched_status.append(file_id)
+        elif len(md5_mwfrs_for_file) == 1:
+            md5_final_status = md5_mwfrs_for_file[0]["final_status"]
+            if md5_final_status == "complete":
+                not_switched_status.append(file_id)
+            elif md5_final_status in ["failed", "stopped"]:
+                problems.append(file_id)
```

Comment:
Consider making `problems` a dict and then adding more informative error messages to report here and below.